core/client/db/src/lib.rs

```diff
@@ -886,10 +790,14 @@ impl<Block> client::backend::Backend<Block, Blake2Hasher> for Backend<Block> whe
 				drop(leaves);
 			}
 
+			for (hash, number, is_best, is_finalized) in meta_updates {
```

Comment:
Just curious - why do we need to replace one loop (inside `if !operation.finalized_blocks.is_empty() {`) with two (inside `operation.pending_block.is_some()` and `operation.pending_block.is_none()`)? Does it fix something?

Reply:
1. In case case `finalized_blocks` are specified metadata for these should be set before the `pending_block`
2. All metadata should be updated after the transaction is committed.